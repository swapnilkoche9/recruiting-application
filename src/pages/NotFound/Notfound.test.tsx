import { render, screen } from '@testing-library/react'

import { QueryClient, QueryClientProvider } from 'react-query'
import { Route, Routes, MemoryRouter } from 'react-router-dom'
import PageNotFound from '.'

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      // âœ… turns retries off
      retry: false,
    },
  },
})

function setup() {
  return render(
    <MemoryRouter>
      <QueryClientProvider client={queryClient}>
        <Routes>
          <Route path='/*' element={<PageNotFound />} />
        </Routes>
      </QueryClientProvider>
    </MemoryRouter>,
  )
}

describe('Not Found Page', () => {
  it('should render page ', async () => {
    const { container } = setup()
    await screen.findByText('page not found')

    expect(container).toMatchSnapshot()
  })
})
