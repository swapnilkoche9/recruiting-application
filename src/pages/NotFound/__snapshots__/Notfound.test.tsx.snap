// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page should render page  1`] = `
<div>
  <div
    class="conatiner"
  >
    <h1>
      page not found
    </h1>
    <a
      href="/"
    >
      go to Applications
    </a>
  </div>
</div>
`;
